### ISSUE TITLE
Send commands from txt files to console or notebook

### ISSUE THREAD

**Issue by pgensler** [jupyterlab/jupyterlab#4031]
I understand that Jupyter is aimed to be for python, but it would be extremely helpfull to have things like a documentation pane if you wanted to view the docs in conjunction with your scripts.

I think it would also much more sense to have the ability to link the text editor with the Console or Notebook, so you could easily send your commands interactively.

**Comment by saulshanabrook** [comment]
> It would be extremely helpful to have things like a documentation pane if you wanted to view the docs in conjunction with your scripts. Especially for R's docs and readthedocs

You might already know this, but you can open an Inspector to view the docs for the current function you are typing in the Notebook, with CMD-I or by right clicking on the cell:
<img width="1076" alt="screen shot 2018-02-27 at 11 37 25 am" src="https://user-images.githubusercontent.com/1186124/36741250-9eb5daf0-1bb2-11e8-9463-a7a2eab64550.png">

**Comment by pgensler** [comment]
I knew about that, but does this support docs for R documentation as well?

**Comment by blink1073** [comment]
Yes, here is a screenshot using the R kernel.  Note that if you right click in an R file, you can select "Create Console for Editor".

<img width="895" alt="image" src="https://user-images.githubusercontent.com/2096628/36748448-7d0f1c26-1bbd-11e8-8e14-4ad593d8023e.png">

**Comment by pgensler** [comment]
Interesting, when I hit crtl-I to view the docs, I get nothing:
![image](https://user-images.githubusercontent.com/7308115/36762098-aa39d5ae-1be7-11e8-8cc3-5f2ac96f135e.png)

**Comment by blink1073** [comment]
The docs are triggered as you type.

**Comment by pgensler** [comment]
In RStudio you can view docs for functions with fn+f1 when your cursor is inside a word, like `paste` in the above case. Could that be a feature in Jupyter Labs?

**Comment by blink1073** [comment]
That might be possible with some changes to the inspector.

**Comment by afshin** [comment]
@pgensler 
> Interesting, when I hit crtl-I to view the docs, I get nothing:
> ![image](https://user-images.githubusercontent.com/7308115/36762098-aa39d5ae-1be7-11e8-8cc3-5f2ac96f135e.png)

The inspector is specifically tied to a kernel. So it'll work for consoles and notebooks that are backed by a kernel that has implemented inspect replies (which includes R). In this screenshot, you were writing code in a text file.
